### PR TITLE
update build

### PR DESCRIPTION
**Pull request by pjfanning**
* scala 2.13.12
* don't hardcode full scala versions in CI yml (easier maintenance)
* use more sbt-pekko-build style dependency management

### CODE REVIEW

File: slick/src/main/scala/org/apache/pekko/projection/slick/internal/SlickProjectionImpl.scala [hunk 1]
```diff
@@ -198,7 +198,7 @@ private[projection] class SlickProjectionImpl[Offset, Envelope, P <: JdbcProfile
       databaseConfig.db.run(offsetStore.saveOffset(projectionId, offset)).map(_ => Done)
 
     private[projection] def newRunningInstance(): RunningProjection =
-      new SlickRunningProjection(RunningProjection.withBackoff(() => mappedSource(), settings), this)
+      new SlickRunningProjection(RunningProjection.withBackoff(() => this.mappedSource(), settings), this)
```
Comment (mdedetrich) [inline slick/src/main/scala/org/apache/pekko/projection/slick/internal/SlickProjectionImpl.scala:201]: Whats reason behind this change, some deprecation warning?

Comment (pjfanning) [reply slick/src/main/scala/org/apache/pekko/projection/slick/internal/SlickProjectionImpl.scala:201]: build issue after updating to scala 2.13.12

```
[error] /home/runner/work/incubator-pekko-projection/incubator-pekko-projection/slick/src/main/scala/org/apache/pekko/projection/slick/internal/SlickProjectionImpl.scala:201:70: reference to mappedSource is ambiguous;
[error] it is both defined in the enclosing class SlickProjectionImpl and inherited in the enclosing class SlickInternalProjectionState as method mappedSource (defined in class InternalProjectionState)
[error] In Scala 2, symbols inherited from a superclass shadow symbols defined in an outer scope.
[error] Such references are ambiguous in Scala 3. To continue using the inherited symbol, write `this.mappedSource`.
[error] Or use `-Wconf:msg=legacy-binding:s` to silence this warning. [quickfixable]
[error]       new SlickRunningProjection(RunningProjection.withBackoff(() => mappedSource(), settings), this)
[error]                                                                      ^
[error] one error found
[error] (slick / Compile / compileIncremental) Compilation failed
```

**Review by mdedetrich — APPROVED** [review]
lgtm

File: project/PekkoSnapshotRepository.scala [hunk 2]
```diff
@@ -19,11 +19,6 @@ object PekkoSnapshotRepositories extends AutoPlugin {
   // If using a snapshot version of either Pekko or Pekko Connectors, add both snapshot repos
   // in case there are transitive dependencies to other snapshot artifacts
   override def projectSettings: Seq[Def.Setting[_]] = {
-    resolvers ++= (sys.props
-      .get("build.pekko.version")
-      .orElse(sys.props.get("build.connectors.kafka.version")) match {
-      case Some(_) => Seq(Resolver.ApacheMavenSnapshotsRepo)
-      case None    => Seq.empty
-    })
+    resolvers += Resolver.ApacheMavenSnapshotsRepo
```
Comment (raboof) [inline project/PekkoSnapshotRepository.scala:22]: Maybe also update(/remove?) the comment above

Comment (pjfanning) [reply project/PekkoSnapshotRepository.scala]: I've changed this with https://github.com/apache/incubator-pekko-projection/pull/107/commits/59f2d1c569f2eca8dd9d2d4b7199fe93fdf3a91f

File: build.sbt [hunk 3]
```diff
@@ -15,6 +15,8 @@ sourceDistName := "apache-pekko-projection"
 sourceDistIncubating := true
 
 ThisBuild / pekkoInlineEnabled := false
+// we want to be able to test with snapshot versions of Pekko dependencies
+ThisBuild / resolvers += Resolver.ApacheMavenSnapshotsRepo
```
Comment (raboof) [inline build.sbt:19]: it was nice that this used to be conditional, but not a big issue to have it here I suppose

**Review by raboof — APPROVED** [review]

**Pull request merged**